ckanext/xloader/tests/test_utils.py

```diff
@@ -0,0 +1,70 @@
+
+
+@pytest.mark.parametrize("input_url, ckan_site_url, xloader_site_url, is_altered, expected", [
+    ("https://ckan.example.org/resource/789", "https://ckan.example.org", "https://xloader.example.org", True, "https://xloader.example.org/resource/789"),
+    ("https://ckan.example.org/resource/789", "https://ckan.example.org", "http://127.0.0.1:3012", True, "http://127.0.0.1:3012/resource/789"),
+    ("https://ckan.example.org/dataset/data", "https://ckan.example.org", "https://xloader.example.org", True, "https://xloader.example.org/dataset/data"),
+    ("https://ckan.example.org/resource/123?foo=bar", "https://ckan.example.org", "https://xloader.example.org", True, "https://xloader.example.org/resource/123?foo=bar"),
+    ("https://ckan.example.org/dataset/456#section", "https://ckan.example.org", "https://xloader.example.org", True, "https://xloader.example.org/dataset/456#section"),
+    ("https://other-site.com/resource/999", "https://ckan.example.org", "https://xloader.example.org", False, ""),
+    ("https://ckan.example.org/resource/123?param=value&other=123", "https://ckan.example.org", "https://xloader.example.org", True, "https://xloader.example.org/resource/123?param=value&other=123"),
+    ("https://ckan.example.org/resource/partial#fragment", "https://ckan.example.org", "https://xloader.example.org", True, "https://xloader.example.org/resource/partial#fragment"),
+    ("https://ckan.example.org/path/to/data?key=value#section", "https://ckan.example.org", "https://xloader.example.org", True, "https://xloader.example.org/path/to/data?key=value#section"),
+    ("https://ckan.example.org/path/to/data?key=value#section", "https://ckan.example.org", "http://localhost:3000", True, "http://localhost:3000/path/to/data?key=value#section"),
+    ("https://ckan.example.org/blackListedPathToS3HostOrigin?key=value#section", "https://ckan.example.org", "https://xloader.example.org", False, ""),
+    ("ftp://ckan.example.org/dataset/456#section", "https://ckan.example.org", "https://xloader.example.org", False, ""),
+    ("https://ckan.example.org/dataset/456#section", "https://ckan.example.org", "", False, ""),
+    ("", "http://127.0.0.1:5000", None, False, ""),
+    ("", "http://127.0.0.1:5000", "", False, ""),
+    (None, "http://127.0.0.1:5000", None, False, ""),
+    (None, "http://127.0.0.1:5000", "", False, ""),
+])
+def test_modify_input_url(input_url, ckan_site_url, xloader_site_url, is_altered, expected):
+    with patch.dict(toolkit.config,
+                    {"ckan.site_url": ckan_site_url,
+                     "ckanext.xloader.site_url": xloader_site_url,
+                     "ckanext.xloader.site_url_ignore_path_regex": "(/blackListedPathToS3HostOrigin|/anotherpath)"}):
+        response = utils.modify_input_url(input_url)
+        if is_altered:
+            assert response == expected
+        else:
+            assert response == input_url
+
+
+
+def test_modify_input_url_no_xloader_site():
```

Comment:
Why is this a separate test instead of just another set of parameterised inputs?

Reply:
it's also in the parameterised, but it may still be nice to keep seperate for 'clarity'